### PR TITLE
Fix cleanup of higher order bytes prior to store.

### PR DESCRIPTION
**Pull request by chriseth**
Fixes https://github.com/ethereum/solidity/issues/1306

### CODE REVIEW

File: libsolidity/codegen/LValue.cpp [hunk 1]
```diff
@@ -231,20 +231,14 @@ void StorageItem::storeValue(Type const& _sourceType, SourceLocation const& _loc
...
 					<< (u256(0x1) << (256 - 8 * dynamic_cast<FixedBytesType const&>(*m_dataType).numBytes()))
 					<< Instruction::SWAP1 << Instruction::DIV;
-			else if (
-				m_dataType->category() == Type::Category::Integer &&
-				dynamic_cast<IntegerType const&>(*m_dataType).isSigned()
-			)
+			else
 				// remove the higher order bits
 				m_context
 					<< (u256(1) << (8 * (32 - m_dataType->storageBytes())))
```
Comment (pirapira) [inline libsolidity/codegen/LValue.cpp:237]: I prefer an assertion `m_dataType->storageBytes() < 32`.  Currently, the storage bytes looks just not 32, either bigger or smaller.

Comment (chriseth) [reply libsolidity/codegen/LValue.cpp:237]: Please take a look at line 203 :-)

Comment (pirapira) [reply libsolidity/codegen/LValue.cpp:237]: Ah, OK.

**Review by pirapira — APPROVED** [review]
The change makes sense to me.

**Pull request merged**